### ISSUE TITLE
The application did not respond

### ISSUE THREAD

**Issue by Stefanuk12** [jackson-nestelroad/panda-discord#2]
I currently have a subcommand group with many subcommands. 

- There is an issue where the bot fails to respond if the group's permissions are on "Everyone" and the subcommand's permissions are on "Owner" or any other permission.
- There's another issue where if the permissions are reversed, you get "Permission Denied." - even though the command permission is "Everyone"

**Comment by jackson-nestelroad** [comment]
I am not able to reproduce the first case. I did this to verify:
- `/permission-test` - permission = Everyone
- `/permission-test group` - permission = Everyone
- `/permission-test group everyone` - permission = Everyone
- `/permission-test group owner` - permission = Owner

I verified that a non-owner could call `/permission-test group everyone` but not `/permission-test group owner`. Could you provide information on how to reproduce the issue you are talking about?

For the second case, I believe this is intended behavior. If the command group's permission is "Owner", then all subcommands of that group are disallowed for everyone except the owner. Permissions should be as general as possible at the top level.

**Comment by Stefanuk12** [comment]
Could we talk over Discord? It would be a lot easier and quicker to diagnose the issue. I've provided files below of a testing bot you can use.

[test2.zip](https://github.com/jackson-nestelroad/panda-discord/files/11414880/test2.zip)

**Comment by jackson-nestelroad** [comment]
What is the expected behavior of the command you have linked? The bot should only respond to `/testing first` if the user is the bot owner. It doesn't make sense for the bot to respond to anyone on `/testing` because it is a command with subcommands. Only the subcommands have handlers that are executed for a response.

Here is my example code:
```ts
class OwnerSubcommand extends SimpleCommand<ExampleBot> {
    public name = 'owner';
    public description = 'Permission test for owner.';
    public category = CommandCategory.Inherit;
    public permission = CommandPermission.Owner;

    public async run({ src }: CommandParameters) {
        await src.send({ content: 'Success!' });
    }
}

class EveryoneSubcommand extends SimpleCommand<ExampleBot> {
    public name = 'everyone';
    public description = 'Permission test for everyone.';
    public category = CommandCategory.Inherit;
    public permission = CommandPermission.Everyone;

    public async run({ src }: CommandParameters) {
        await src.send({ content: 'Success!' });
    }
}

export class PermissionTestCommand extends NestedCommand<ExampleBot> {
    public name = 'permission-test';
    public description = 'Permission test.';
    public category = CommandCategory.Fun;
    public permission = CommandPermission.Everyone;
    public cooldown = StandardCooldowns.Low;

    public subcommands = [EveryoneSubcommand, OwnerSubcommand];
}
```

In this example:
- Everyone can use `/permission-test everyone`.
- Only the owner can use `/permission-test owner`.

**Comment by Stefanuk12** [comment]
In the command I have linked, an error saying that I do not have permission is expected. When testing out your code, I have the same issue. However, based on your reply, this might be the intended behaviour. If it is, a message saying I do not have permission would be helpful and confuse users less.

**Comment by jackson-nestelroad** [comment]
Ah, I know what you mean. You're talking about the actual "Application did not respond" message when the bot chooses to ignore an interaction. Yes, this is a bit misleading. I've added an ephemeral reply to slash commands when a user does not meet the required permissions. Message commands will still be ignored to avoid spam. I'll push this change up to NPM.

**Comment by jackson-nestelroad** [comment]
The fix should be up in `panda-discord@4.4.0`. Let me know if there are any other issues around this. Thanks!

<img width="466" alt="image" src="https://user-images.githubusercontent.com/40396320/236706120-f37c4509-c4e7-4275-809a-5456fa61a20b.png">
